packages/dm-core/src/types.ts

```diff
@@ -92,3 +92,30 @@ export type TPlugin = {
 }
 
 export type TUserIdMapping = { userId: string; username: string }
+
+// View config specific
+
+export type TViewConfig = {
+  type: string
+  scope: string
+}
+
+export type TReferenceViewConfig = TViewConfig & {
+  recipe: string
+}
+
+export type TRecipeViewConfig = TViewConfig & {
+  recipe: TUiRecipe
+}
+
+export function isReferenceViewConfig(
```

Comment:
Don't you only need one of these type predicates?
If its not one, it's always the other